pwndbg/typeinfo.py

```diff
@@ -119,18 +120,28 @@ def load(name):
 {name} foo;
 '''.format(**locals())
 
-    filename = '%s/%s_%s' % (tempdir, arch, '-'.join(name.split()))
+    filename = '%s/%s_%s.cc' % (tempdir, arch, '-'.join(name.split()))
 
-    if not os.path.exists(filename + '.o'):
-        with open(filename + '.cc', 'w+') as f:
-            f.write(source)
-            f.flush()
+    with open(filename, 'w+') as f:
+        f.write(source)
+        f.flush()
 
+    compile(filename)
+
+    return gdb.lookup_type(name)
+
+def compile(filename=None, address=0):
+    """Compile and extract symbols from specified file"""
+    if filename is None:
+        print("Specify a filename to compile.")
+        return
+
+    objectname = filename.split(".")[0] + ".o"
+
+    if not os.path.exists(objectname):
         gcc     = pwndbg.gcc.which()
-        gcc    += ['-w','-c','-g',filename + '.cc','-o',filename + '.o']
+        gcc    += ['-w', '-c', '-g', filename, '-o', objectname]
         subprocess.check_output(' '.join(gcc), shell=True)
 
     with pwndbg.events.Pause():
-        gdb.execute('add-symbol-file %s.o 0' % filename, from_tty=False, to_string=True)
-
-    return gdb.lookup_type(name)
+        gdb.execute('add-symbol-file %s %s' % (objectname, address), from_tty=False, to_string=True)
```

Comment:
While you're adding functionality for this, would you mind breaking this bit out into a new function?